serialization.go

```diff
@@ -315,14 +315,19 @@ func InferAcceptHeaderFromType(ans any) string {
 }
 
 func parseAcceptHeader(accept string, ans any) string {
-	if strings.Index(accept, ",") > 0 {
-		accept = accept[:strings.Index(accept, ",")]
-	}
-	if accept == "*/*" {
-		accept = ""
-	}
 	if accept == "" {
-		accept = InferAcceptHeaderFromType(ans)
+		return InferAcceptHeaderFromType(ans)
+	}
+
+	vals := strings.Split(accept, ",")
+	// if the string contains `*/*` try to infer
+	// the header from the type
+	for _, v := range vals {
+		if strings.Contains(v, "*/*") {
+			return InferAcceptHeaderFromType(ans)
+		}
 	}
```

Comment:
I'm not sure it's the right strategy. 
We should only infer if it contains */* but only if the previous mime types cannot be satisfied.

Reply:
Then I misunderstood our discussion. I’ll close and send another PR

Reply:
Sorry about that, I might have not been very clear :/
Thanks anyway, I think we're near the final result

Reply:
I assume you were expecting something a little more like this:

```
func Send(w http.ResponseWriter, r *http.Request, ans any) (err error) {
	for _, header := range parseAcceptHeader(r.Header) {
		switch inferAcceptHeader(header, ans) {
		case "application/xml":
			err = SendXML(w, nil, ans)
		case "text/html":
			err = SendHTML(w, r, ans)
		case "text/plain":
			err = SendText(w, nil, ans)
		case "application/json":
			err = SendJSON(w, nil, ans)
		case "application/x-yaml", "text/yaml; charset=utf-8", "application/yaml": // https://www.rfc-editor.org/rfc/rfc9512.html
			err = SendYAML(w, nil, ans)
		}

		if err != nil {
			continue
			// should we log and continue?
			// or
			// if err == nil we just return?
		}
		return nil
	}

	if err != nil {
		return err
	}
	return errors.New("no supported Accept header was not provided from: " + r.Header.Get("Accept"))
}

```

Reply:
Yeah so this is where we to define what behavior wit expect. So when the user asks for a HTML route using `application/json`. The above ends with:

```
func TestSend(t *testing.T) {
	const templateName = "template"
	template, err := template.New(templateName).Parse("Hello World")
	require.NoError(t, err)

	w := httptest.NewRecorder()
	r := httptest.NewRequest(http.MethodGet, "/", nil)
	r.Header.Set("Accept", "application/json")
	Send(w, r, &StdRenderer{
		templates:         template,
		templateToExecute: templateName,
	})
	body := w.Body.String()
	require.Equal(t, "{}\n", body)
	require.Equal(t, "application/json", w.Header().Get("Content-Type"))
}

```

I will open this as  PR. It would probably be easier to define test cases for what we expect here.

Your last point doesn't really seem to be a problem though right? Like if the header is `application/json, */*` you're gonna end up with the same error above. Now if the caller is asking for `application/json, application/xml` that's kinda on the caller, right now we would probably return the error of not being able to encode xml in that case.